Mathlib/Combinatorics/SimpleGraph/Finsubgraph.lean

```diff
@@ -59,21 +61,45 @@ instance : Sup G.Finsubgraph :=
 instance : Inf G.Finsubgraph :=
   ⟨fun G₁ G₂ => ⟨G₁ ⊓ G₂, G₁.2.subset inter_subset_left⟩⟩
 
+@[simp, norm_cast] lemma coe_bot : (⊥ : G.Finsubgraph) = (⊥ : G.Subgraph) := rfl
+
+@[simp, norm_cast]
+lemma coe_sup (G₁ G₂ : G.Finsubgraph) : ↑(G₁ ⊔ G₂) = (G₁ ⊔ G₂ : G.Subgraph) := rfl
+
+@[simp, norm_cast]
+lemma coe_inf (G₁ G₂ : G.Finsubgraph) : ↑(G₁ ⊓ G₂) = (G₁ ⊓ G₂ : G.Subgraph) := rfl
+
 instance : DistribLattice G.Finsubgraph :=
   Subtype.coe_injective.distribLattice _ (fun _ _ => rfl) fun _ _ => rfl
 
-instance [Finite V] : Top G.Finsubgraph :=
-  ⟨⟨⊤, finite_univ⟩⟩
+section Finite
+variable [Finite V]
+
+instance instTop : Top G.Finsubgraph where top := ⟨⊤, finite_univ⟩
+instance instSupSet : SupSet G.Finsubgraph where sSup s := ⟨⨆ G ∈ s, ↑G, Set.toFinite _⟩
+instance instInfSet : InfSet G.Finsubgraph where sInf s := ⟨⨅ G ∈ s, ↑G, Set.toFinite _⟩
+
+@[simp, norm_cast] lemma coe_top : (⊤ : G.Finsubgraph) = (⊤ : G.Subgraph) := rfl
```

Comment:
I'd be tempted to write the `\u`s explicitly here, just to make the naming easy to justify.